### PR TITLE
blockchain_db: speedup tx output gathering

### PR DESCRIPTION
**Pull request by moneromooo-monero**
We know all the data we'll want for getblocks.bin is contiguous

### CODE REVIEW

**Comment by moneromooo-monero** [comment]
I did time this one, the time taken by the outer "fetch all the data" loop went down from about 20 time units to about 13 time units (when all was cached)

File: src/rpc/core_rpc_server.cpp [hunk 1]
```diff
@@ -252,34 +252,37 @@ namespace cryptonote
...
+      if (n_txes_to_lookup > 0)
+      {
+        std::vector<std::vector<uint64_t>> indices;
+        bool r = m_core.get_tx_outputs_gindexs(req.no_miner_tx ? bd.second.front().first : bd.first.second, n_txes_to_lookup, indices);
         if (!r)
         {
           res.status = "Failed";
           return false;
         }
+        if (indices.size() != n_txes_to_lookup || res.output_indices.back().indices.size() != (req.no_miner_tx ? 1 : 0))
```
Comment (stoffu) [inline src/rpc/core_rpc_server.cpp]: The latter condition seems to always hold by construction and seems unrelated to the result of `get_tx_outputs_gindexs`.

Comment (moneromooo-monero) [reply src/rpc/core_rpc_server.cpp]: Yes, and if it does not hold, the call errors out.
